### DISCUSSION TITLE
Airflow with TLS-enabled Sentinel as a broker for celery doesn't work

### DISCUSSION THREAD

**Discussion by TomYork1986** [apache/airflow#19510] [General]
Hello,

We're trying to enable TLS for celery in our Airflow 2.2.1, but it doesn't work. What we have:
Redis with Sentinel configured with TLS and work fine:
_redis.conf:_
```
port 0
tls-port 6379
tls-cert-file "<path_to_file>"
tls-key-file "<path_to_file>"
tls-ca-cert-file "<path_to_file>"
tls-auth-clients no
tls-replication yes
```
_sentinel.conf_
```
port 0
tls-port 26379
tls-cert-file "<path_to_file>"
tls-key-file "<path_to_file>"
tls-ca-cert-file "<path_to_file>"
tls-auth-clients no
tls-replication yes
```
Airflow celery workers and scheduler are fine with Redis as a broker:
```
...
[celery]
broker_url = redis://<username>:<password>@<host1>:6379;redis://<username>:<password>@<host2>:6379;redis://<username>:<password>@<host3>:6379
ssl_active = True
ssl_key = <path_to_file>
ssl_cert = <path_to_file>
ssl_cacert = <path_to_file>
```
But when we try to use Sentinel as a broker:
```
broker_url = sentinel://<username>:<password>@<rhost_1>:26379;sentinel://<username>:<password>@<rhost_2>:26379;sentinel://<username>:<password>@<host_3>:26379
```
We have the following in celery.err:
> [2021-11-10 10:39:02,783: ERROR/MainProcess] consumer: Cannot connect to sentinel://default:**@<host1>:26379//: No master found for 'redis-cluster'.
Trying again in 32.00 seconds... (16/100)

> [2021-11-10 10:39:34,819: ERROR/MainProcess] consumer: Cannot connect to sentinel://default:**@<host2>:26379//: No master found for 'redis-cluster'.
Trying again in 32.00 seconds... (16/100)

And in sentinel.log:
> 19332:X 10 Nov 2021 10:39:02.782 # Error accepting a client connection: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
> 19332:X 10 Nov 2021 10:39:34.818 # Error accepting a client connection: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol

P.S.
we had to make a trick to avoid "The broker you configured does not support SSL_ACTIVE to be True" error in the airflow/config_templates/default_celery.py file:
```
try:
    if celery_ssl_active:
        if 'amqp://' in broker_url:
            broker_use_ssl = {
                'keyfile': conf.get('celery', 'SSL_KEY'),
                'certfile': conf.get('celery', 'SSL_CERT'),
                'ca_certs': conf.get('celery', 'SSL_CACERT'),
                'cert_reqs': ssl.CERT_REQUIRED,
            }
        elif 'redis://' in broker_url:
            broker_use_ssl = {
                'ssl_keyfile': conf.get('celery', 'SSL_KEY'),
                'ssl_certfile': conf.get('celery', 'SSL_CERT'),
                'ssl_ca_certs': conf.get('celery', 'SSL_CACERT'),
                'ssl_cert_reqs': ssl.CERT_REQUIRED,
            }
        elif 'sentinel://' in broker_url:
            broker_use_ssl = {
                'ssl_keyfile': conf.get('celery', 'SSL_KEY'),
                'ssl_certfile': conf.get('celery', 'SSL_CERT'),
                'ssl_ca_certs': conf.get('celery', 'SSL_CACERT'),
                'ssl_cert_reqs': ssl.CERT_REQUIRED,
            }
        else:
            raise AirflowException(
                'The broker you configured does not support SSL_ACTIVE to be True. '
                'Please use RabbitMQ or Redis if you would like to use SSL for broker.'
            )
```

Does it mean that the latest version of Airflow doesn't support Sentinel with TLS?

**Comment by potiuk** [comment]
Airlfow supports "plain" redis as queue. Sentinel has never been tested (not in CI). So it is likely that some fixes might be necessary. If you would like to make it work also with sentinel, maybe you can make it work and contribute a fix back?

**Reply by TomYork1986** [reply]
With great pleasure, but I am afraid that my programmer skills are not enough for this :(
Perhaps it makes sense to create a feature request?

**Reply by potiuk** [reply]
Perhaps. Maybe someone will also be interested. Just to let you know -  since it is an open-source project, it's developed by contributors like you, so if there will be no-one who has access and interest in supporting sentinel (and ability to test it), it's unlikely to be implemented. So the most "sure" way to get it implemented is if you do it yourself.

**Reply by TomYork1986** [reply]
Got it! I will try to get our dev team involved in this issue.
Thank you!

**Reply by potiuk** [reply]
All contributions like that are welcome :)
